packages/dataviews/src/dataviews-layouts/grid/index.tsx

```diff
@@ -157,9 +157,15 @@ function GridItem< Item >( {
 				justify="space-between"
 				className="dataviews-view-grid__title-actions"
 			>
-				<div { ...clickableTitleItemProps } { ...titleA11yProps }>
-					{ renderedTitleField }
-				</div>
+				<ItemClickWrapper
+					item={ item }
+					isItemClickable={ isItemClickable }
+					onClickItem={ onClickItem }
+					LinkComponent={ LinkComponent }
+					className="dataviews-view-grid__title-field dataviews-title-field"
+				>
+					<div { ...titleA11yProps }>{ renderedTitleField }</div>
```

Comment:
Do we need this `div`? Shouldn't the props be on the `temClickWrapper`?

Additionally if the passed component is a `button` we would have invalid HTML.

Reply:
Right something is not right here :)